### DISCUSSION TITLE
请问Hidden应用场景有哪些？为什么Hidden还要保留 校验字段属性呢？

### DISCUSSION THREAD

**Discussion by ByYogi** [ant-design/ant-design#44392] [Q&A] [answered]
> React 里 hidden input 没啥意义，把这个标记记在一个变量（或状态）里就可以了。 https://github.com/ant-design/ant-design/issues/34670

**Comment by zombieJ** [comment]
好巧，正好改的这个例子用到了：https://github.com/ant-design/ant-design/pull/44391

在 v3 时期，Form 的 store 值总是存在里面。到了 v4 则是注册过的字段才会存在，v4 之后的 Form 更贴近原生 form 的逻辑。对于不存在 Field 的数据需要通过 `getFieldsValue(true)` 获取。像上面这个 PR 就是，不需要存在用户交互，但是需要保留字段数据的一个使用场景。

**Reply by ByYogi** [reply, marked as answer]
https://stackblitz.com/edit/react-pgh6nl?file=demo.tsx

哦，是为了保留字段让`Form`在`onFinish`时能检测到。
比如`users`字段是动态生成的，不能使用`name`受控。但是又需要这个动态生成的数据可以传入到`onFinish`。就`setFieldsValue`保存到`Form`中，再放一个对应的`FormItem`但是`Hidden`。这样`onFinish`就可以拿到这个users动态数据了
